### DISCUSSION TITLE
Plugin configuration

### DISCUSSION THREAD

**Discussion by helmecke** [marlonrichert/zsh-launchpad#5] [Q&A] [answered]
Hi @marlonrichert, thank you for this nice template. When I put my `zsh-users/zsh-syntax-highlighting` configuration in any of the `rc.d` files it does not work. It only works if I put it in `.zshrc`. Could you please explain why.

Config snippet:
```zsh
ZSH_HIGHLIGHT_HIGHLIGHTERS=(main regexp)
typeset -A ZSH_HIGHLIGHT_REGEXP
ZSH_HIGHLIGHT_REGEXP+=('\bsudo\b' fg=yellow,bold,underline)
```

**Comment by marlonrichert** [comment, marked as answer]
Inside a function, `typeset -A` is the same as `local -A`. To create a global parameter, use `typeset -gA` instead:
```zsh
typeset -ga ZSH_HIGHLIGHT_HIGHLIGHTERS=(main regexp)
typeset -gA ZSH_HIGHLIGHT_REGEXP=('\bsudo\b' fg=yellow,bold,underline)
```

For more info, see the following sections in the Zsh manual:
* [Local parameters](https://zsh.sourceforge.io/Doc/Release/Parameters.html#Local-Parameters)
* [`typeset`](https://zsh.sourceforge.io/Doc/Release/Shell-Builtin-Commands.html#index-typeset)

**Reply by marlonrichert** [reply]
@helmecke PS: I maintain all of my software for Zsh in my free time. If you enjoy using it, please consider giving back by [sponsoring me](https://github.com/sponsors/marlonrichert). 🙂
